### PR TITLE
fix: unused vars rule

### PR DESCRIPTION
**Pull request by sang4lv**
##### Checklist

- [ x ] `npm test` passes
- [ x ] tests and/or benchmarks are included
- [ x ] documentation is changed or added
- [ x ] commit message follows commit guidelines

##### Description of change

add `no-unused-vars` rule which is prerequisite for rule `@typescript-eslint/no-unused-vars`.
New test case is included, CI config updated.

### CODE REVIEW

**Comment by whxaxes** [comment]
8.0.1

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 sudo: false
 language: node_js
 node_js:
-  - '8'
```
Comment (atian25) [inline .travis.yml]: @吖猩 这个包的 8.0 这个之前是不支持 Node 8 么？

Comment (whxaxes) [reply .travis.yml]: 支持呀

Comment (atian25) [reply .travis.yml]: 那这里干掉单测是不是有问题

**Comment by popomore** [comment]
为啥要去掉这个？

**Pull request merged**